Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -1260,38 +1278,75 @@ namespace Multiplayer
         {
             AZLOG_WARN("Attempting to enable autonomous control for an invalid entity");
             return;
         }
 
-        entityHandle.GetNetBindComponent()->SetOwningConnectionId(connectionId);
-        if (connectionId == InvalidConnectionId)
+        for (NetworkEntityHandle subEntity : entityList)
         {
-            entityHandle.GetNetBindComponent()->SetAllowAutonomy(true);
+            if (NetBindComponent* netbindComponent = subEntity.GetNetBindComponent())
+            {
+                netbindComponent->SetOwningConnectionId(connectionId);
+
+                if (connectionId == InvalidConnectionId)
+                {
+                    netbindComponent->SetAllowAutonomy(true);
+                }
+            }
         }
+    }
 
-        auto* hierarchyComponent = entityHandle.FindComponent<NetworkHierarchyRootComponent>();
-        if (hierarchyComponent != nullptr)
+    void MultiplayerSystemComponent::OnRootSpawnableReady([[maybe_unused]] AZ::Data::Asset<AzFramework::Spawnable> rootSpawnable, [[maybe_unused]] uint32_t generation)
+    {
+        // Spawn players waiting to be spawned. This can happen when a player connects before a level is loaded, so there isn't any player spawner components registered
+        IMultiplayerSpawner* spawner = AZ::Interface<IMultiplayerSpawner>::Get();
+        if (!spawner)
         {
-            for (AZ::Entity* subEntity : hierarchyComponent->GetHierarchicalEntities())
+            AZLOG_ERROR("Attempting to spawn players on level load failed. No IMultiplayerSpawner found. Ensure one is registered.");
+            return;
+        }
+
+        for (const auto& playerWaitingToBeSpawned : m_playersWaitingToBeSpawned)
+        {
+            auto preActivationCallback = [playerWaitingToBeSpawned](const INetworkEntityManager::EntityList& entityList)
             {
-                NetworkEntityHandle subEntityHandle = NetworkEntityHandle(subEntity);
-                NetBindComponent* subEntityNetBindComponent = subEntityHandle.GetNetBindComponent();
+                const ConnectionId connectionId = playerWaitingToBeSpawned.connection ? playerWaitingToBeSpawned.connection->GetConnectionId() : InvalidConnectionId;
+                EnableAutonomousControl(entityList, connectionId);
+            };
 
-                if (subEntityNetBindComponent != nullptr)
-                {
-                    subEntityNetBindComponent->SetOwningConnectionId(connectionId);
-                    if (connectionId == InvalidConnectionId)
-                    {
-                        subEntityNetBindComponent->SetAllowAutonomy(true);
-                    }
-                }
+            NetworkEntityHandle controlledEntity = spawner->OnPlayerJoin(playerWaitingToBeSpawned.userId, playerWaitingToBeSpawned.agent, preActivationCallback);
+            if (!controlledEntity.Exists())
+            {
+                AZLOG_WARN("Attempting to spawn players on level load failed.");
```

Comment:
I would consider printing additional details here, such as `userId`.

Reply:
I'd also indicate that it's a Multiplayer level to disambiguate from other spawning logs.

Reply:
good idea, fixed!